### PR TITLE
Add application server sets and media relay sets to Brand

### PR DESCRIPTION
**Pull request by danigargar**
#### Type Of Change 
- [ ] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [x] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by ironArt3mis — CHANGES_REQUESTED** [review]
Functional review required

File: library/Ivoz/Provider/Domain/Model/Brand/Brand.php [hunk 1]
```diff
@@ -52,27 +53,38 @@ public function getId(): ?int
...
-                $targetFeatues = [
+//        Assertion::notEmpty(
+//            $this->getRelApplicationServerSets(),
+//            'ApplicationServerSets cannot be empty'
+//        );
+//
+//        Assertion::notEmpty(
+//            $this->getRelMediaRelaySets(),
+//            'MediaRelaySets cannot be empty'
+//        );
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Domain/Model/Brand/Brand.php:65]: please no comments

File: library/Ivoz/Provider/Domain/Service/ApplicationServerSetsRelBrand/AvoidDeleteAllByBrand.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return;
+        }
+
+        $isEmptyApplicationServerSets = empty($brand->getRelApplicationServerSets());
+        if (!$isEmptyApplicationServerSets) {
+            return;
+        }
+
+        throw new \DomainException(
+            'Brand cannot be left without a corresponding Application Server Sets'
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Domain/Service/ApplicationServerSetsRelBrand/AvoidDeleteAllByBrand.php:35]: 'Application Server Sets cannot be empty'``

File: library/Ivoz/Provider/Domain/Service/Brand/AvoidEmptyApplicationServerSets.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public function execute(BrandInterface $brand): void
+    {
+        $relApplicationServerSets = $brand->getRelApplicationServerSets();
+
+        if (!empty($relApplicationServerSets)) {
+            return;
+        }
+
+        throw new \DomainException(
+            'Application Server Sets cannot be empty'
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Domain/Service/Brand/AvoidEmptyApplicationServerSets.php:28]: 'Rel Application Server Sets cannot be empty'

File: library/Ivoz/Provider/Domain/Service/MediaRelaySetsRelBrand/AvoidDeleteAllByBrand.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            return;
+        }
+
+        $isEmptyMediaRelaySets = empty($brand->getRelMediaRelaySets());
+        if (!$isEmptyMediaRelaySets) {
+            return;
+        }
+
+        throw new \DomainException(
+            'Brand cannot be left without a corresponding Media Relay Set'
```
Comment (rbatistadev) [inline library/Ivoz/Provider/Domain/Service/MediaRelaySetsRelBrand/AvoidDeleteAllByBrand.php:38]: 'Media Relay Sets cannot be empty'

File: library/Ivoz/Provider/Domain/Service/Brand/AvoidEmptyApplicationServerSets.php (hunk 3 above)
Comment (danigargar) [reply library/Ivoz/Provider/Domain/Service/Brand/AvoidEmptyApplicationServerSets.php:28]: Actually what is exposed is applicationServerSets, that's why I think its better this way.

**Review by rbatistadev — APPROVED** [review]

**Review by ironArt3mis — APPROVED** [review]

**Pull request merged**